solana/programs/matching-engine/src/processor/auction/execute_fast_order/mod.rs

```diff
@@ -240,3 +95,210 @@ fn handle_execute_fast_order<'info>(
         order_executed_event,
     })
 }
+
+pub struct ActiveAuctionAccountInfos<'ix> {
+    pub auction_best_offer_token: AccountInfo<'ix>,
+    pub auction_executor_token: AccountInfo<'ix>,
+    pub auction_initial_offer_token: AccountInfo<'ix>,
+    pub auction_custody_token: AccountInfo<'ix>,
+    pub active_auction: AccountInfo<'ix>,
+    pub auction_custodian: AccountInfo<'ix>,
+}
+
+pub fn get_user_amount_and_new_status_and_penalized(
+    auction: &Auction,
+    custody_token: &TokenAccount,
```

Comment:
These are the deserialized data of `ActiveAuctionAccountInfos::active_auction` and `ActiveAuctionAccountInfos::auction_custody_token` right? Maybe consolidate these with that struct (sort of like how you made the helper struct for settle auction none)